### PR TITLE
Migrate cluster dns test to sig network

### PR DESCRIPTION
**Pull request by guangxuli**
**What this PR does / why we need it**:

Just migrate dns relevant e2e test files to sig network. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Ref Umbrella issue #49161

**Special notes for your reviewer**:

**Release note**:
```release-note
none
```

### CODE REVIEW

File: test/e2e/network/example_cluster_dns.go [hunk 1]
```diff
@@ -156,3 +159,14 @@ var _ = framework.KubeDescribe("ClusterDns [Feature:Example]", func() {
 func getNsCmdFlag(ns *v1.Namespace) string {
 	return fmt.Sprintf("--namespace=%v", ns.Name)
 }
+
+// pass enough context with the 'old' parameter so that it replaces what your really intended.
```
Comment (m1093782566) [inline test/e2e/network/example_cluster_dns.go]: Could you please explain why move this function here?

**Comment by guangxuli** [comment]
@thockin updated the `BUILD `file, PTAL, thanks.

**Pull request merged**